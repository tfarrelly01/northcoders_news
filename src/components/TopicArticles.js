import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchUsers, fetchTopicArticles } from '../actions/asyncActions';
import ArticleList from './ArticleList';

class TopicArticles extends React.Component {
	componentDidMount() {
		this.props.fetchUsers();
		this.props.fetchTopicArticles(this.props.match.params.topic_slug);
	}

	componentWillReceiveProps(nextProps) {
		if (this.props.match.params.topic_slug !== nextProps.match.params.topic_slug) {
			this.props.fetchTopicArticles(nextProps.match.params.topic_slug);
		}
	}

	render() {
    return (
      <ArticleList
        articles={this.props.articles} 
        users={this.props.users}
      />
		);
	}
}

const mapStateToProps = (state) => {
	return {
		users: state.users,
		articles: state.articles
	};
}

const mapDispatchToProps = (dispatch) => {
	return { 
		fetchUsers: () => {
			dispatch(fetchUsers());
		}, 
		fetchTopicArticles: (id) => {
			dispatch(fetchTopicArticles(id));
		}
	};
}

TopicArticles.propTypes = {
	users: PropTypes.array.isRequired,
	articles: PropTypes.array.isRequired,
	match: PropTypes.object.isRequired,
	fetchUsers: PropTypes.func.isRequired,
	fetchTopicArticles: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(TopicArticles);
