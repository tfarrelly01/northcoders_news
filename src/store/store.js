import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../reducer/reducer';

const logger = createLogger();

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk, logger),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
      ? window.devToolsExtension()
      : f => f
  )
);

export default store;