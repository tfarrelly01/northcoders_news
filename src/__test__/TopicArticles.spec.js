import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../store/store';

import TopicArticles from '../components/TopicArticles';

describe('TopicArticles Component', () => {
  test('TopicArticles renders correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <TopicArticles />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});