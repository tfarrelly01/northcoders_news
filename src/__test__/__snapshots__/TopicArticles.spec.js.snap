// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicArticles Component TopicArticles renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(TopicArticles) />,
  "nodes": Array [
    <Connect(TopicArticles) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(TopicArticles) />
      </Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(TopicArticles) />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(TopicArticles) />,
        "_debugID": 2,
        "_renderedOutput": <Connect(TopicArticles) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(TopicArticles) />
  </Provider>,
}
`;
